paasta_tools/paasta_maintenance.py

```diff
@@ -422,6 +424,43 @@ def status():
     print "%s" % status.text
 
 
+def is_safe_to_kill(hostname):
+    """Checks if a host has drained or reached its maintenance window
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    if is_host_drained(hostname) or hostname in get_hosts_in_maintenance_window():
+        return True
+    return False
+
+
+def is_host_drained(hostname):
+    """Checks if a host has drained successfully by confirming it is
+    draining and currently running 0 tasks
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    mesos_state = get_mesos_state_from_leader()
+    task_counts = get_mesos_task_count_by_slave(mesos_state)
+    slave_task_count = task_counts[hostname].count
+    if is_host_draining(hostname=hostname) and slave_task_count == 0:
+        return True
+    return False
+
+
+def get_hosts_in_maintenance_window():
```

Comment:
This name is a bit deceptive. A maintenance window has a start time and a duration (which effectively gives it an end time). This function doesn't care if we are past the end time of the maintenance window.